vowpalwabbit/multiclass.cc

```diff
@@ -129,14 +129,14 @@ void print_score(vw& all, example& ec, uint32_t prediction)
   std::stringstream label_ss;
   label_ss << ec.l.multi.label;
 
-  all.sd->print_update(all.holdout_set_off, all.current_pass, label_ss.str(), pred_ss.str(), ec.num_features,
-      all.progress_add, all.progress_arg);
+  all.sd->print_update(all.ec.holdout_set_off, all.gs.current_pass, label_ss.str(), pred_ss.str(), ec.num_features,
```

Comment:
@jackgerrits feels like we could implement a different version of print which already has access to all.ec and all.oc to the very least. I think we could even get rid of the reference to all.gs. Instead of sending things back and forth just set once and call from everywhere else.

Reply:
To add to this it would be interesting to think how this functionality should work for a library - are we just printing to a different output device? Or we updating some state and printing is one possible thing we do with it?